File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamcloud/callback/SingleFileInfoCallback.java
package in.dragonbra.javasteam.steam.handlers.steamcloud.callback;

import in.dragonbra.javasteam.enums.EResult;
import in.dragonbra.javasteam.protobufs.steamclient.SteammessagesClientserverUfs.CMsgClientUFSGetSingleFileInfoResponse;
import in.dragonbra.javasteam.steam.handlers.steamcloud.SteamCloud;
import in.dragonbra.javasteam.steam.steamclient.callbackmgr.CallbackMsg;
import in.dragonbra.javasteam.types.JobID;

import java.util.Date;

/**
 * This callback is received in response to calling {@link SteamCloud#getSingleFileInfo(int, String)}.
 */
public class SingleFileInfoCallback extends CallbackMsg {

    private EResult result;

    private int appID;

    private String fileName;

    private byte[] shaHash;

    private Date timestamp;

    private int fileSize;

    private boolean isExplicitDelete;

    public SingleFileInfoCallback(JobID jobID, CMsgClientUFSGetSingleFileInfoResponse.Builder msg) {
        setJobID(jobID);

        result = EResult.from(msg.getEresult());

        appID = msg.getAppId();
        fileName = msg.getFileName();
        shaHash = msg.getShaFile().toByteArray();
        timestamp = new Date(msg.getTimeStamp() * 1000L);
        fileSize = msg.getRawFileSize();
        isExplicitDelete = msg.getIsExplicitDelete();
    }

    /**
     * @return the result of the request.
     */
    public EResult getResult() {
        return result;
    }

    /**
     * @return the App ID the file is for.
     */
    public int getAppID() {
        return appID;
    }

    /**
     * @return the file name request.
     */
    public String getFileName() {
        return fileName;
    }

    /**
     * @return the SHA hash of the file.
     */
    public byte[] getShaHash() {
        return shaHash;
    }

    /**
     * @return the timestamp of the file.
     */
    public Date getTimestamp() {
        return timestamp;
    }

    /**
     * @return the size of the file.
     */
    public int getFileSize() {
        return fileSize;
    }

    /**
     * @return if the file was explicitly deleted by the user.
     */
    public boolean isExplicitDelete() {
        return isExplicitDelete;
    }
}

File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamcloud/callback/SingleFileInfoCallback.kt
package `in`.dragonbra.javasteam.steam.handlers.steamcloud.callback

import `in`.dragonbra.javasteam.enums.EResult
import `in`.dragonbra.javasteam.protobufs.steamclient.SteammessagesCloudSteamclient.CCloud_GetSingleFileInfo_Response
import `in`.dragonbra.javasteam.steam.handlers.steamcloud.SteamCloud
import `in`.dragonbra.javasteam.steam.handlers.steamunifiedmessages.callback.ServiceMethodResponse
import java.util.*

/**
 * This callback is received in response to calling [SteamCloud.getSingleFileInfo].
 */
@Suppress("MemberVisibilityCanBePrivate")
class SingleFileInfoCallback(response: ServiceMethodResponse<CCloud_GetSingleFileInfo_Response.Builder>) {

    /**
     * Gets the result of the request.
     */
    val result: EResult = response.result

    /**
     * Gets the App ID the file is for.
     */
    val appID: Int

    /**
     * Gets the file name request.
     */
    val fileName: String

    /**
     * Gets the SHA hash of the file.
     */
    val shaHash: ByteArray

    /**
     * Gets the timestamp of the file.
     */
    val timestamp: Date

    /**
     * Gets the size of the file.
     */
    val fileSize: Int

    /**
     * Gets if the file was explicitly deleted by the user.
     */
    val isExplicitDelete: Boolean

    init {
        val msg = response.body

        appID = msg.appId
        fileName = msg.fileName
        shaHash = msg.shaFile.toByteArray()
        timestamp = Date(msg.timeStamp * 1000L)
        fileSize = msg.rawFileSize
        isExplicitDelete = msg.isExplicitDelete
    }
}
